data/lib/core/storages.lua

```diff
@@ -17,3 +17,26 @@ PlayerStorageKeys = {
 
 GlobalStorageKeys = {
 }
+
+-- Checking for duplicate storages:
+local function extractValues(tab, ret)
+	if type(tab) == "number" then
+		table.insert(ret, tab)
+	else
+		for _, v in pairs(tab) do
+			extractValues(v, ret)
+		end
+	end
+end
```

Comment:
This function is unnecessarily complicated, it gives false impression of being recursive.
There's no reason it has to call itself. It's only used by passing two tables, so for readability, this would be much better:
```suggestion
local function extractValues(tab, ret)
	for _, v in pairs(tab) do
 		table.insert(ret, tab)
	end
end
```
In fact, this function does really need to be a function at all if there is no intention to reuse it.
Don't overcomplicate the code.

Reply:
Thats what happened when someone copy a code that does something in one base and move to another where it doesn't make sense without realizing what exactly was he doing. @DSpeitchert the function makes sense when you have nested storage tables, then the recursion would be needed, what is not tfs base case. For simple tables your code is more than enough.